typescript/src/deposit.ts

```diff
@@ -179,7 +183,8 @@ export async function submitDepositTransaction(
  *          - the deposit transaction in the raw format
  */
 export async function assembleDepositTransaction(
-  deposit: Deposit,
+  deposit: DepositScriptParameters,
+  amount: BigNumber,
```

Comment:
I'm not a huge fan of the detached amount but I don't see a better way 🤷 

Also, a new question arises: Can we refactor the whole `Deposit` interface to operate on wallet and refund key hashes instead of plain text? Of course, this is a huge change and not for this PR but this may be worth exploring if we observe that we have a constant need to make changes like the one presented in this PR. After all, the initial decision to base the `Deposit` interface on plain text public keys could be wrong as we were lacking a lot of information back in the day.

Reply:
Refactor of the `Deposit` interface: https://github.com/keep-network/tbtc-v2/pull/441